ppa/archive/eebo_tcp.py

```diff
@@ -0,0 +1,68 @@
+"""
+Code for working with EEBO-TCP (Text Creation Partnership) content.
+"""
+
+from eulxml import xmlmap
+
+
+class Page(xmlmap.XmlObject):
+    """A page of content in an EEBO-TCP text"""
+
+    # not sure how these are used; repeat in pairs? page label (?)
+    ref = xmlmap.StringField("@REF")
+    #: page number (optional)
+    number = xmlmap.StringField("@N")
+
+    # parent div type =~ page type / label ?
+    section_type = xmlmap.StringField("ancestor::DIV1/@TYPE")
+
+    # page breaks delimit content instead of containing it;
+    # use following axis to find all text nodes following this page break
+    text_contents = xmlmap.StringListField("following::text()")
+
+    def __repr__(self):
+        return f"<Page {self.label} ({self.section_type})>"
+
+    def page_contents(self):
+        "generator of text strings between this page break and the next one"
+
+        # strictly speaking we are returning lxml "smart strings"
+        # (lxml.etree._ElementUnicodeResult)
+
+        # iterate and yield text following the current page
+        # break until we hit the next page break
+        for i, text in enumerate(self.text_contents):
+            # lxml handles text between elements as "tail" text;
+            # the parent of the tail text is the preceding element
+            if text.is_tail:
+                # if text precedes a GAP tag, include the display content
+                # from the DISP attribute (for now)
+                if text.getparent().tag == "GAP":
+                    yield text.getparent().get("DISP")
```

Comment:
Is this because GAP tags aren't text tags? Are there other tags that we'll need to account?

Reply:
Correct, there's no text content in GAP tags, it's only in the attributes.

I didn't encounter any and don't remember any, but it's been a long time since I did much TEI; I will do some querying and check for other tags in case there are others we need to handle.